src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -58,7 +58,7 @@ internal static class DefaultQueryHandler
 
             // execute query
             QueryResult result;
-            var executor = context.GetApiService<IQueryExecutor>();
+            var executor = context.GetApiContextService<IQueryExecutor>();
```

Comment:
@rayao Could we save instance creation/dispose here? For example, as ApiContext may have multiple InnovacationContext, then in case of invocationContext.getService, it may get an object created and dispose several  times, but if we use invocationContext,parentContext.getService, only one object of one type is needed to be created?